src/main/java/org/example/controllers/UsersController.java

```diff
@@ -53,6 +68,12 @@ public ResponseEntity<String> assignDepartmentToUser(@PathVariable int userId, @
         return ResponseEntity.ok("Department assigned to user successfully");
```

Comment:
such logic should be on the service layer, not in the controller. Controller should not know anything about logic what should happen if some variable is null or not